frontend/src/pages/wifiPage/WifiPageContainer.tsx

```diff
@@ -42,6 +48,34 @@ export default ({
       .catch(() => setIsConnected(false));
   };
 
+  const waitForAdvancedConfigUrlTimeout = 700;
+
+  const waitForAdvancedConfigUrl = () => {
+    const interval = setInterval(async () => {
+      try {
+        getVncWpaGuiUrl()
+          .then((data: any) => {
+            if (data.url !== "") {
+              clearInterval(interval)
+              setAdvancedConfigUrl(data.url);
+            }
+          })
+      } catch (_) {}
+    }, waitForAdvancedConfigUrlTimeout);
+  }
+
+  const startAdvancedWifiConfig = () => {
+    startVncWpaGui()
+      .then(() => setTimeout(waitForAdvancedConfigUrl, 300))
+      .catch(() => setAdvancedConfigError(true))
+  };
+
+  const stopAdvancedWifiConfig = () => {
+    setAdvancedConfigUrl("")
+    stopVncWpaGui()
+      .catch(() => setAdvancedConfigError(true))
+  };
+
```

Comment:
I think you should move these methods for vnc starting, stopping and polling into the Advanced(Wifi)ConfigDialog itself, to improve it's isolation and reduce duplication. they could be in an extra container for that component (we've been using these less religiously recently, but they're still sometimes nice).